src/test/java/xbot/common/injection/wpi_factories/TestCommonLibFactory.java

```diff
@@ -19,18 +21,30 @@ public void makeOneOfEverything() {
         clf.createEncoder("foo", 1, 2, 1);
         clf.createDigitalInput(1);
         clf.createAnalogInput(1);
-        clf.createXboxController(1);
+        clf.createXboxController(2);
         clf.createSolenoid(1);
-        clf.createDigitalOutput(1);
+        clf.createDigitalOutput(3);
         clf.createServo(1);
-        clf.createSpeedController(1);
+        clf.createSpeedController(2);
         clf.createCANTalon(1);
         clf.createGyro();
         clf.createLidarLite(I2C.Port.kOnboard);
         clf.createAdvancedJoystickButton(j, 1);
         clf.createAnalogHIDButton(j, 1, -1, 1);
-        clf.createGamepad(1, 10);
+        clf.createGamepad(3, 10);
         
         clf.createHeadingModule(pf.createPIDManager("bar", 1, 0, 0));
     }
+    
+    @Test
+    public void doubleAllocate() {
+        clf.createCANTalon(1);
+        
+        try {
```

Comment:
Rather than catch the exception, you can add an annotation which tells JUnit to expect an exception and fail if it isn't thrown.

Reply:
Ooh, nice. I'll look that up.